### DISCUSSION TITLE
Enhancement request: sort by name and number

### DISCUSSION THREAD

**Discussion by wwhio** [Byron/dua-cli#128] [Ideas]
It will be nice to have these features in normal and interactive mode.

Sort by name:

```
chapter1.txt
chapter10.txt
chapter11.txt
chapter2.txt
chapter20.txt
chapter3.txt
image1.jpg
image10.jpg
image2.jpg
image3.jpg
```

Sort by name and number:

```
chapter1.txt
chapter2.txt
chapter3.txt
chapter10.txt
chapter11.txt
chapter20.txt
image1.jpg
image2.jpg
image3.jpg
image10.jpg
```

I think "sort by name" is much easier to implement and faster than "sort by name and number".

But it will be nice to have both.

reference: https://blog.miigon.net/posts/how-to-sort-file-names-naturally/

**Comment by Byron** [comment]
If you like you could flesh this out in the context of `dua` here. How would that look like in an example? Maybe that way it can get to a point where you or somebody interested can submit a PR according to the specification developed here.

**Reply by wwhio** [reply]
Thank you. I will do that.

**Reply by wwhio** [reply]
I have changed the description. Do you think it's good? Or more details are needed?

**Reply by Byron** [reply]
Thanks, that looks descriptive enough.
